src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -232,6 +251,66 @@ ExtensionBooleanResponse handleIndicesModuleNameRequest(IndicesModuleRequest ind
         return indicesModuleNameResponse;
     }
 
+    /**
+     * Handles a request to update a setting from OpenSearch.  Extensions must register their setting keys and consumers within the settingUpdateConsumer map
+     *
+     * @param updateSettingsRequest  The request to handle.
+     * @return A response acknowledging the request.
+     */
+    @SuppressWarnings("unchecked")
+    ExtensionBooleanResponse handleUpdateSettingsRequest(UpdateSettingsRequest updateSettingsRequest) {
+        logger.info("Registering UpdateSettingsRequest received from OpenSearch");
+
+        boolean settingUpdateStatus = true;
+
+        WriteableSetting.SettingType settingType = updateSettingsRequest.getSettingType();
+        Setting<?> componentSetting = updateSettingsRequest.getComponentSetting();
+        Object data = updateSettingsRequest.getData();
+
+        // Setting updater in OpenSearch performs setting change validation, only need to cast the consumer to the corresponding type and
+        // invoke the consumer
```

Comment:
Not a request to change anything in your PR but a note for awareness.  The original validator and parser are lost in the existing setup, so if there was some extra validation step (for example, minimum value integer) it might not properly validate on the OpenSearch side.   That's fine for now as it will eventually be handled by #154.